### PR TITLE
Feature/admin invite user

### PR DESCRIPTION
**Pull request by peggles2**
Ensure admin can invite user.
- Created user is set to approved instead of pending email verification
- user does not need a password
- Validations to pass
- fixtures updated to approved user.
- fix rspec tests
- fix cucumber tests
- update email flow

### CODE REVIEW

File: features/step_definitions/user_steps.rb [hunk 1]
```diff
@@ -24,14 +24,14 @@
...
     if hash[:password_updated_at]
-     user = User.find_by_email(hash[:email])
-     user.update_attributes(password_updated_at: hash[:password_updated_at].to_date)
+      user = User.find_by_email(hash[:email])
+      user.update_attributes(password_updated_at: hash[:password_updated_at].to_date)
     end
   end
 end
 
 When /^(?:I|they) click the complete registration link in the email$/ do
```
Comment (MothOnMars) [inline features/step_definitions/user_steps.rb]: I'm puzzled why we're changing from the "complete registration" link to the "sites" link. Don't we still want them to visit the "complete registration" page and fill out their agency?

File: spec/mailers/emailer_spec.rb [hunk 2]
```diff
@@ -156,8 +155,7 @@
     let(:user) do
       mock_model(User,
            :email => "invitee@agency.com",
-           :contact_name => 'Invitee Joe',
-           :email_verification_token => 'some_special_token')
+           :contact_name => 'Invitee Joe')
```
Comment (MothOnMars) [inline spec/mailers/emailer_spec.rb]: if time, update to 1.9 hash syntax

File: spec/mailers/emailer_spec.rb [hunk 3]
```diff
@@ -167,7 +165,7 @@
 
     it { should deliver_to("invitee@agency.com") }
     it { should have_subject(/\[Search.gov\] Welcome to Search.gov/) }
-    it { should have_body_text(/https:\/\/localhost:3000\/complete_registration\/some_special_token\/edit/) }
```
Comment (MothOnMars) [inline spec/mailers/emailer_spec.rb]: see question above - seems like we should still be directing them to the `complete_registration` page

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -430,47 +430,54 @@
...
         expect(new_user.is_affiliate?).to be true
         expect(new_user.requires_manual_approval).to be false
-        expect(new_user.is_pending_email_verification?).to be true
+        expect(new_user.is_approved?).to be true
         expect(new_user.welcome_email_sent).to be false
         expect(affiliate.users).to include(new_user)
       end
 
-      it "should receive welcome new user added by affiliate email verification" do
+      it 'should receive welcome new user added by affiliate email verification' do
```
Comment (MothOnMars) [inline spec/models/user_spec.rb]: If time, omit the `should`: `'receives the welcome new user...`

File: features/step_definitions/user_steps.rb (hunk 1 above)
Comment (peggles2) [reply features/step_definitions/user_steps.rb]: That complete registration just passes a token. Yes we will later change that back. But for this release we are just going to go to the sites page.

**Pull request merged**